File: stacks/java/commons/src/test/java/co/com/bancolombia/utils/PercentageTest.java
package co.com.bancolombia.utils;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class PercentageTest {

    @Test
    public void shouldGenerateTenPercentage() {
        // Arrange
        int percentage = 50;
        float max = Percentage.MAX_PERCENTAGE * 1f;
        int real = 0;
        int tolerance = 10;
        // Act
        for (int i = 0; i < max; i++) {
            if (Percentage.isInPercentage(percentage)) {
                real++;
            }
        }
        float realPercentage = real / max;
        // Assert
        float maxPercentage = (percentage + tolerance) / max;
        System.out.println(realPercentage * Percentage.MAX_PERCENTAGE + "% of max " + maxPercentage * Percentage.MAX_PERCENTAGE + "%");
        assertThat(realPercentage).isLessThanOrEqualTo(maxPercentage);
    }
}

File: stacks/nodejs/commons/lib/percentage.test.js
const {isInPercentage} = require('./percentage');

test('should calculate percentage', () => {
    // Arrange
    const percentage = 50;
    const max = parseFloat(100);
    const tolerance = 10;
    let real = 0;
    // Act
    for (let i = 0; i < max; i++) {
        if (isInPercentage(percentage)) {
            real++;
        }
    }
    // Assert
    console.log(real)
    const realPercentage = real / max;
    const maxPercentage = (percentage + tolerance) / max;
    console.log(realPercentage * max + "% of max " + maxPercentage * max + "%");
    expect(realPercentage).toBeLessThan(maxPercentage);
});
